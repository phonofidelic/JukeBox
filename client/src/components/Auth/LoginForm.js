import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import { 
	Button,
	TextField,
	Typography
} from '@material-ui/core';
import { validation } from '../../utils';

const form = reduxForm({
	form: 'loginForm'
});

const renderField = ({
 input, 
 label, 
 meta: { touched, error }, 
 children, 
 ...custom 
}) => (
	<TextField
		label={Boolean(touched && error) ? error : label}
		error={Boolean(touched && error)}
		{...input}
		{...custom} 
	/>
);

export class LoginForm extends Component {
	render() {
		const { 
			auth,
			handleLogin,
			handleSubmit,
		} = this.props;


		const styles = {
			root: {
				marginTop: '50px'
			},
			header: {
				margin: '10px'
			}
		}

		// console.log('LoginForm, auth.loginErr:', Boolean(auth.loginErr))
		return (
			<form
				style={styles.root}
				onSubmit={handleSubmit(handleLogin)}
				auth={auth}
			>
				<div style={styles.header}>
					<Typography>
		        Sign in:
		      </Typography>
				</div>
				<div>
					<Field 
						id="signin-email"
						component={renderField}
						type="email"
						name="email"
						label="Email"
						aria-label="Email"
						validate={[validation.required, validation.email]}
						placeholder="Enter your email"
						data-cy="signin-email"
					/>
				</div>
				<div>
					<Field
						id="signin-password"
						component={renderField}
						type="password"
						name="password"
						label="Password"
						aria-label="Password"
						validate={[validation.required]}
						placeholder="Enter your password"
						data-cy="signin-password"
					/>
				</div>
				<div style={{marginTop: '20px'}}>
					<Button 
						variant="outlined"
						type="submit"
						data-cy="signin-button"
					>
						Sign in
					</Button>
				</div>
			</form>
		);
	}
}

LoginForm.propTypes = {
	auth: PropTypes.object.isRequired,
	handleLogin: PropTypes.func.isRequired,
	handleSubmit: PropTypes.func.isRequired,
};

export default form(LoginForm);
