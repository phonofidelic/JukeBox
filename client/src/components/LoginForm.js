import React, { Component } from 'react';
import { Field, reduxForm } from 'redux-form';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { validation } from '../utils';

const form = reduxForm({
	form: 'loginForm'
});

const renderField = ({
 input, 
 label, 
 meta: { touched, error }, 
 children, 
 ...custom 
}) => (
	<TextField
		label={Boolean(touched && error || custom.loginerr) ? error : label}
		error={Boolean(touched && error)}
		{...input}
		{...custom} 
	/>
);

class LoginForm extends Component {
	render() {
		const { 
			auth,
			handleLogin,
			handleSubmit,
			handleClearError,
		} = this.props;


		const styles = {
			root: {
				marginTop: '50px'
			},
			header: {
				margin: '10px'
			}
		}

		console.log('LoginForm, auth.loginErr:', Boolean(auth.loginErr))
		return (
			<form
				style={styles.root}
				onSubmit={handleSubmit(handleLogin)}
				onChange={auth.loginErr && handleClearError}
				auth={auth}
			>
				<div style={styles.header}>
					<Typography>
		        Sign in:
		      </Typography>
				</div>
				<div>
					<Field 
						component={renderField}
						type="email"
						name="email"
						label="Email"
						loginerr={auth.loginErr}
						validate={[validation.required, validation.email]}
						placeholder="Enter your email"
					/>
				</div>
				<div>
					<Field 
						component={renderField}
						type="password"
						name="password"
						label="Password"
						loginerr={auth.loginErr}
						validate={[validation.required]}
						placeholder="Enter your password"
					/>
				</div>
				<div style={{marginTop: '20px'}}>
					<Button 
						variant="outlined"
						type="submit"
					>
						Sign in
					</Button>
				</div>
			</form>
		);
	}
}

export default form(LoginForm);