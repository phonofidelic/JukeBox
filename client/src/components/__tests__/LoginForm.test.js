
import React from 'react';
// import ReactDOM from 'react-dom';
// import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { reduxForm } from 'redux-form';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { LoginForm } from '../LoginForm'; 

/***
 *	Solution for snapshot testing decorated redux-form:
 *	https://github.com/erikras/redux-form/issues/849#issuecomment-255412223
 */

describe('LoginForm', () => {
	let props = {
		auth: {loginErr: false},
		handleLogin: jest.fn(),
		handleSubmit: jest.fn()
	}
	
	const fakeStore = createStore(() => ({}));
	const DecoratedForm = reduxForm({ form: 'testForm' })(LoginForm);

	it('renders correctly', () => {
	  const tree = renderer
	    .create(<Provider store={fakeStore}><DecoratedForm {...props} /></Provider>)
	    .toJSON();

	  expect(tree).toMatchSnapshot();
	});
});

// describe('LoginForm', () => {
// 	let props;
// 	let mountedLoginForm;

// 	const loginForm = () => {
// 		if (!mountedLoginForm) {
// 			mountedLoginForm = mount(
// 				<LoginForm {...props} />
// 			);
// 		}

// 		return mountedLoginForm;
// 	}

// 	beforeEach(() => {
// 		props = {
// 			auth: undefined,
// 			handleLogin: undefined,
// 			handleSubmit: jest.fn()
// 		}
// 	});

// 	mountedLoginForm = undefined;

// 	it('always renders a form', () => {
// 		const forms = loginForm().find('form');
// 		expect(forms.length).toBeGreaterThan(0);
// 	});
// });


