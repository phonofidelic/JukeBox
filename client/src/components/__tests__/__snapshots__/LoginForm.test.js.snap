// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
<form
  auth={
    Object {
      "loginErr": false,
    }
  }
  onSubmit={undefined}
  style={
    Object {
      "marginTop": "50px",
    }
  }
>
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <p
      className="MuiTypography-root-1 MuiTypography-body1-10"
    >
      Sign in:
    </p>
  </div>
  <div>
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-25"
      onBlur={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-34 MuiInputLabel-root-29 MuiInputLabel-formControl-30 MuiInputLabel-animated-33"
        data-shrink={false}
        htmlFor={undefined}
      >
        Email
      </label>
      <div
        className="MuiInput-root-39 MuiInput-formControl-40 MuiInput-underline-43"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-47 MuiInput-inputType-50"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder="Enter your email"
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="email"
          value=""
        />
      </div>
    </div>
  </div>
  <div>
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-25"
      onBlur={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-34 MuiInputLabel-root-29 MuiInputLabel-formControl-30 MuiInputLabel-animated-33"
        data-shrink={false}
        htmlFor={undefined}
      >
        Password
      </label>
      <div
        className="MuiInput-root-39 MuiInput-formControl-40 MuiInput-underline-43"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-47 MuiInput-inputType-50"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder="Enter your password"
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="password"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "marginTop": "20px",
      }
    }
  >
    <button
      className="MuiButtonBase-root-68 MuiButton-root-52 MuiButton-outlined-56"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="submit"
    >
      <span
        className="MuiButton-label-53"
      >
        Sign in
      </span>
      <span
        className="MuiTouchRipple-root-71"
      />
    </button>
  </div>
</form>
`;
